viewing_party/party.py

```diff
@@ -1,23 +1,119 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    '''
+    Create a dict containing info about one movie, with "title", "genre", and "rating", as keys, mapped to values passed in as arguments.
+    '''
+    if not title or not genre or not rating:
+        return None
 
+    movie = {}
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+
+    return movie
+
+def add_to_watched(user_data, movie):
+    '''
+    Add a movie to a user's list of watched movies
+    '''
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    '''
+    Add a movie to a user's watchlist
+    '''
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    '''
+    Move a movie from watchlist to watched
+    '''
+    for movie in user_data["watchlist"]:
+        if movie["title"] == movie_title:
+            watched_movie = movie
+            user_data["watched"].append(watched_movie)
+            user_data["watchlist"].remove(watched_movie)
+    
+    return user_data

```

Comment:
This works in this particular case because we only expect the user to have one copy of each movie in their watchlist. But in general, try to avoid modifying a data structure as we're doing a `for` loop over it; very strange and hard to debug behavior can result. Consider instead using the loop to identify what needs to be removed, then remove it after the loop.